ts/index.ts

```diff
@@ -0,0 +1,216 @@
+    this.logger = inputs.getString("logger");
+    this.gitHubToken = inputs.getString("github-token");
+    // extra-tags is deprecated but we still honor it
+    this.extraLabels =
+      inputs.getString("extra-labels") === ""
+        ? inputs.getString("extra-tags")
+        : "";
+    this.spdxExpression = inputs.getString("spdx-expression");
+    this.errorOnConflict = inputs.getBool("error-on-conflict");
+    this.includeOutputPaths = inputs.getBool("include-output-paths");
+    this.flakeHubPushBinary = inputs.getStringOrNull("flakehub-push-binary");
+  }
+
+  private verifyVisibility(): Visibility {
+    const visibility = inputs.getString("visibility");
+    if (!VISIBILITY_OPTIONS.includes(visibility)) {
+      throw new Error(
+        `Visibility option \`${visibility}\` not recognized. Available options: ${VISIBILITY_OPTIONS.join(", ")}.`,
+      );
+    }
+    return visibility as Visibility;
+  }
+
+  private async executionEnvironment(): Promise<ExecutionEnvironment> {
+    const env: ExecutionEnvironment = {};
+
+    env.FLAKEHUB_PUSH_VISIBLITY = this.visibility;
+    env.FLAKEHUB_PUSH_TAG = this.tag;
+    env.FLAKEHUB_PUSH_HOST = this.host;
+    env.FLAKEHUB_PUSH_LOG_DIRECTIVES = this.logDirectives;
+    env.FLAKEHUB_PUSH_LOGGER = this.logger;
+    env.FLAKEHUB_PUSH_GITHUB_TOKEN = this.gitHubToken;
+    env.FLAKEHUB_PUSH_NAME = this.flakeName;
+    env.FLAKEHUB_PUSH_REPOSITORY = this.repository;
+    env.FLAKEHUB_PUSH_DIRECTORY = this.directory;
+    env.FLAKEHUB_PUSH_GIT_ROOT = this.gitRoot;
+    env.FLAKEHUB_PUSH_EXTRA_LABELS = this.extraLabels;
+    env.FLAKEHUB_PUSH_SPDX_EXPRESSION = this.spdxExpression;
+
+    if (!this.rolling) {
+      env.FLAKEHUB_PUSH_ROLLING = "false";
+    }
+
+    if (!this.mirror) {
+      env.FLAKEHUB_PUSH_MIRROR = "false";
+    }
+
+    if (!this.errorOnConflict) {
+      env.FLAKEHUB_PUSH_ERROR_ON_CONFLICT = "false";
+    }
+
+    if (!this.includeOutputPaths) {
+      env.FLAKEHUB_PUSH_INCLUDE_OUTPUT_PATHS = "false";
+    }
+
+    if (this.rollingMinor !== null) {
+      env.FLAKEHUB_PUSH_ROLLING_MINOR = this.rollingMinor.toString();
+    }
+
+    return env;
+  }
+
+  private get flakeName(): string {
+    let name: string;
+
+    const org = process.env["GITHUB_REPOSITORY_OWNER"];
+    const repo = process.env["GITHUB_REPOSITORY"];
+
+    if (this.name !== null) {
+      if (this.name === "") {
+        throw new Error("The `name` field can't be an empty string");
+      }
+
+      const parts = this.name.split("/");
+
+      if (parts.length === 1 || parts.length > 2) {
+        throw new Error("The specified `name` must of the form {org}/{repo}");
+      }
+
+      const orgName = parts.at(0);
+      const repoName = parts.at(1);
+
+      if (orgName !== org) {
+        throw new Error(
+          `The org name \`${orgName}\` that you specified using the \`name\` input doesn't match the actual org name \`${org}\``,
+        );
+      }
```

Comment:
Let's use setFailed instead of throw new Error. Throwing errors will look like a bug.